pandas/core/groupby/frame.py

```diff
@@ -0,0 +1,368 @@
+import collections
+from textwrap import dedent
+from functools import partial
+
+import numpy as np
+
+from pandas.compat import map
+import pandas.core.common as com
+from pandas.core.series import Series
+from pandas.core.generic import _shared_docs
+from pandas.core.groupby.groupby import GroupBy
+from pandas.core.groupby import base
+from pandas.util._decorators import Appender
+from pandas.core.index import MultiIndex, CategoricalIndex
+from pandas.core.arrays.categorical import Categorical
+from pandas.core.frame import DataFrame
+from pandas.core.groupby.groupby import NDFrameGroupBy
+from pandas.core.groupby.series import SeriesGroupBy
+from pandas._libs.lib import count_level_2d
+from pandas.plotting._core import boxplot_frame_groupby
```

Comment:
since stuff is being moved en-masse, this is a good time to clean up imports.  After the standard `__future__`-->stdlib-->3rd party blocks, I advocate blocks in order of dependencies.  So imports from `_libs`, then `util` and `compat`, then `core` (I like to put `core.dtypes` before other `core`), and last `plotting`.  YMMV.

Reply:
updated